lib/search/index.ts

```diff
@@ -211,6 +232,13 @@ export const middleware: S.Middleware = (store) => {
       }
 
       if (
+        titleOnly &&
+        !searchTerms.every((term) =>
+          note.content.split('\n', 1)[0].includes(term.toLocaleLowerCase())
```

Comment:
I don't think that this is going to give us what we want.
we might prefer to trap the searchable content into a separate variable and then keep the rest of the search flow the same

```ts

const searchText = titleOnly ? getNoteTitle(note) : note.content;
…
```

I'm pretty sure I don't have that right, but I mean to point out that we have a function that extracts the title exactly how it's done in the note list and in the note editor and that apart from that the searching is performed identically whether the text is truncated or not.